File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AudienceStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AudienceStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "clientId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AudienceStateEventId value;

    public AudienceStateEventIdFlattenedDto()
    {
        this(new AudienceStateEventId());
    }

    public AudienceStateEventIdFlattenedDto(AudienceStateEventId value)
    {
        this.value = value;
    }

    public AudienceStateEventId toAudienceStateEventId()
    {
        return this.value;
    }


    public String getClientId()
    {
        return this.value.getClientId();
    }

    public void setClientId(String clientId)
    {
        this.value.setClientId(clientId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AudienceStateEventIdFlattenedDto other = (AudienceStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/AudienceStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class AudienceStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'clientId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AudienceStateEventId
     */
    private $value;

    /**
     * @param AudienceStateEventId $value
     */
    public function __construct(AudienceStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AudienceStateEventId();
        }
    }

    /**
     * @return AudienceStateEventId
     */
    public function toAudienceStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getClientId()
    {
        return $this->value->getClientId();
    }

    /**
     * @param string $clientId
     */
    public function setClientId($clientId)
    {
        $this->value->setClientId($clientId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getClientId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return AudienceStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setClientId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

